samples/bottom-navigation/src/androidMain/kotlin/com/slack/circuit/sample/navigation/MainActivity.kt

```diff
@@ -25,24 +36,51 @@ class MainActivity : AppCompatActivity() {
       ?.let { WindowCompat.getInsetsController(it, window.decorView) }
       ?.isAppearanceLightStatusBars = true
 
+    // CircuitX Navigation
+    val interceptors =
+      persistentListOf(AndroidScreenAwareNavigationInterceptor(this), InfoScreenRewriteInterceptor)
```

Comment:
Okay nice! So the idea here is that `AndroidScreenAwareNavigationInterceptor` will remove the need to 
```
val backStack = rememberSaveableBackStack(root = root)
val delegate = rememberCircuitNavigator(backStack)
val navigator = rememberAndroidScreenAwareNavigator(delegate, this@MainActivity)
val interceptingNavigator = rememberCircuitInterceptingNavigator(
    navigator = navigator,
    ...
)
```
if we're already using a `AndroidScreenAwareNavigator`?

Reply:
Correct, using the interceptor provides an alternative to the nested delegation approach.